pallets/runtime/tests/src/bridge.rs

```diff
@@ -12,790 +9,515 @@ use frame_support::{
-    );
-    assert_ok!(Bridge::propose_bridge_tx(charlie, bridge_tx.clone()));
-    assert_tx_approvals_and_next_block!(controller, 0, 2);
-    assert_tx_approvals_and_next_block!(controller, 1, 0);
-
-    assert_eq!(MultiSig::proposal_ids(&controller, proposal), Some(0));
-    let new_alices_balance = Balances::total_balance(&AccountKeyring::Alice.public());
-    assert_eq!(new_alices_balance, alices_balance + amount);
-    // Attempt to handle the same transaction again.
-    assert_eq!(
-        Bridge::bridge_tx_details(AccountKeyring::Alice.public(), &1).status,
-        BridgeTxStatus::Handled
-    );
-    assert_noop!(
-        Bridge::handle_bridge_tx(Origin::signed(controller), bridge_tx),
-        Error::ProposalAlreadyHandled
-    );
+fn alice_bridge_tx(amount: u128) -> BridgeTx {
+    make_bridge_tx(Alice.public(), amount)
+}
+
+fn bob_bridge_tx(amount: u128) -> BridgeTx {
+    make_bridge_tx(Bob.public(), amount)
+}
+
+fn proposal_tx(recipient: Public, amount: u128) -> (BridgeTx, Box<Call>) {
+    let tx = make_bridge_tx(recipient, amount);
+    let call = bridge::Call::handle_bridge_tx(tx.clone());
+    (tx, box Call::Bridge(call))
+}
+
+fn alice_proposal_tx(amount: u128) -> Box<Call> {
+    proposal_tx(Alice.public(), amount).1
+}
+
+fn alice_balance() -> u128 {
+    Balances::total_balance(&Alice.public())
+}
+
+fn alice_tx_details(tx_id: u32) -> BridgeTxDetail {
+    Bridge::bridge_tx_details(Alice.public(), tx_id)
+}
+
+fn proposals(amount: u128) -> Vec<(BridgeTx, Box<Call>)> {
+    [Alice, Eve, Ferdie]
+        .iter()
+        .map(|acc| proposal_tx(acc.public(), amount))
+        .collect()
+}
+
+fn signers_approve_proposal(proposal: Call, signers: &[Public]) -> BridgeTx {
+    let controller = Bridge::controller();
+    match proposal {
+        Call::Bridge(bridge::Call::handle_bridge_tx(ref tx)) => {
+            let mut proposal_id = None;
+
+            // Use minimun number of signs to approve it.
+            for i in 0..(MIN_SIGNS_REQUIRED as usize) {
+                assert_ok!(Bridge::propose_bridge_tx(
+                    Origin::signed(signers[i]),
+                    tx.clone()
+                ));
+
+                // Fetch proposal ID if unknown.
+                if proposal_id.is_none() {
+                    proposal_id = MultiSig::proposal_ids(&controller, proposal.clone());
+                }
+
+                // Verify approvals.
+                assert_eq!(
+                    MultiSig::proposal_detail(&(controller, proposal_id.unwrap())).approvals,
+                    (i + 1) as u64
+                );
+            }
+            return tx.clone();
+        }
+        _ => panic!("Invalid call"),
+    }
+}
+
+/// Advances the system `block_number` and run any scheduled task.
+fn next_block() -> Weight {
+    let block_number = System::block_number() + 1;
+    System::set_block_number(block_number);
+    // Call the timelocked tx handler.
+    Scheduler::on_initialize(block_number)
+}
```

Comment:
Seems like we have logic like this elsewhere in the tests... grep for `next_block` and dedup?